lib/internal/errors.js

```diff
@@ -828,16 +765,21 @@ function setArrowMessage(err, arrowMessage) {
 // Hide stack lines before the first user code line.
 function hideInternalStackFrames(error) {
   overrideStackTrace.set(error, (error, stackFrames) => {
-    let frames = stackFrames;
+    let result = '';
     if (typeof stackFrames === 'object') {
-      frames = ArrayPrototypeFilter(
-        stackFrames,
-        (frm) => !StringPrototypeStartsWith(frm.getFileName() || '',
-                                            'node:internal')
-      );
+      for (const frame of stackFrames) {
```

Comment:
this should probably avoid for-of, both for robustness and for performance. can it use forEach?

Reply:
The loop type won't make a big difference here. I personally like for...of loops. They are easy to read.

Reply:
for-of invokes the iterator protocol, which is often slower than array iteration methods (which are identical to a classic for loop in modern v8)

Reply:
Last time I checked, `for(;;)` is still the fastest way to iterate over an array or array-like object.

Reply:
@aduh95 modern v8 turns array iteration methods into the same underlying code as a for loop, altho that may not apply with primordials. either way, for-of loops should be avoided.

Reply:
for-of is worse than a for loop, because it invokes the iterator protocol, which is slower and less robust.

Reply:
I would like to keep this as is for now.